### PR TITLE
SSL-Handshake implementation

### PR DESCRIPTION
**Pull request by msvinaykumar**
Signed-off-by: msvinaykumar <vinakuma@redhat.com>

### CODE REVIEW

File: src/main/java/com/autotune/utils/GenericRestApiClient.java [hunk 1]
```diff
@@ -116,9 +123,15 @@ public void setAuthHeaderString(String authHeaderString) {
...
      */
-    private JSONObject fetchMetricsJson(String methodType, String queryString) throws IOException {
+    private JSONObject fetchMetricsJson(String methodType, String queryString) throws IOException, NoSuchAlgorithmException, KeyStoreException, KeyManagementException {
         String jsonOutputInString = "";
-        try (CloseableHttpClient httpclient = HttpClients.createDefault()) {
+        SSLContext sslContext = SSLContexts.custom()
+                .loadTrustMaterial((chain, authType) -> true).build();
+        SSLConnectionSocketFactory sslConnectionSocketFactory =
+                new SSLConnectionSocketFactory(sslContext, new String[]
+                        {"SSLv2Hello", "SSLv3", "TLSv1","TLSv1.1", "TLSv1.2" }, null,
```
Comment (chandrams) [inline src/main/java/com/autotune/utils/GenericRestApiClient.java:132]: Can you please check on the protocols list. SSLv3 - I remember had vulnerabilities and is not recommended

Comment (msvinaykumar) [reply src/main/java/com/autotune/utils/GenericRestApiClient.java]: thats right . removed SSLv3.

**Comment by dinogun** [comment]
To add more context on why SSLv3 should be removed -> https://disablessl3.com/

**Comment by dinogun** [comment]
@msvinaykumar It looks like your commits are not GPG key verified, can you please ensure that you have the proper GPG key setup and you sign each of your commits with it, thanks

**Pull request closed without merging**